### PR TITLE
618 events card

### LINKED ISSUE #618: [FRONTEND] Create the events card
**Is your feature request related to a problem? Please describe.**

Image

**BEFORE MERGING**

- [ ] Storybooks created where possible
- [ ] Tested with [nock](https://tanstack.com/query/v3/docs/framework/react/guides/testing) if data fetching
- [ ] Tests written for critical interactions
- [ ] PR Reviewed (For non-trivial changes)
- [ ] Changes tested after rebasing on master or merging in master (_hint_: `git fetch origin master:master`, then `git rebase master` or `git merge master`)
- [ ] All required PR checks passing

### PR DESCRIPTION
**Pull request by zlrkw11**
closes #618

### CODE REVIEW

**Comment by zlrkw11** [comment]
image
- desktop viewport

image
- small viewport

**Review by bcho892 — COMMENTED** [review]
can we have an onclick handler for the button?

**Comment by zlrkw11** [comment]
image
- desktop finalised

image
- mobile finalised

onClick handler added

File: client/src/components/generic/EventsCard/EventsCard.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  title: string
+  location: string
+  content: React.ReactNode
+}
+
+type props = EventsCardProps
+
+const EventsCard = ({ date, title, location, content }: props) => {
+  const handleOnClick = () => {
+    console.log("Button has been clicked")
```
Comment (bcho892) [inline client/src/components/generic/EventsCard/EventsCard.tsx:14]: this should be a prop, not hard coded

File: client/src/components/generic/EventsCard/EventsCard.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  }
+  const Divider = () => {
+    return <div className="bg-gray-3 mb-4 mt-4 h-[1px] w-full"></div>
+  }
+  return (
+    <div className="h-full w-full border text-center md:text-left">
+      <div className="m-4">
```
Comment (bcho892) [inline client/src/components/generic/EventsCard/EventsCard.tsx:21]: padding on parent is better here

File: client/src/components/generic/EventsCard/EventsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        <h5 className="font-bold">{date}</h5>
+
+        <h3 className="text-dark-blue-100 mt-1 font-bold">{title}</h3>
+
+        <div className="text-gray-4 mt-2">{location}</div>
+        <Divider />
+
+        <div className="text-left">{content}</div>
+      </div>
+      <div className="mx-4 my-8">
```
Comment (bcho892) [inline client/src/components/generic/EventsCard/EventsCard.tsx:31]: not needed if parent has padding

File: client/src/components/generic/EventsCard/EventsCard.tsx (hunk 1 above)
Comment (zlrkw11) [reply client/src/components/generic/EventsCard/EventsCard.tsx:14]: fixed

File: client/src/components/generic/EventsCard/EventsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import Button from "../FigmaButtons/FigmaButton"
+
+interface EventsCardProps {
+  date: string
+  title: string
+  location: string
```
Comment (bcho892) [inline client/src/components/generic/EventsCard/EventsCard.tsx:6]: nit: These might be optional?

**Review by bcho892 — CHANGES_REQUESTED** [review]

**Comment by zlrkw11** [comment]
image

image

padding fix & unnecessary divs removed

**Review by bcho892 — APPROVED** [review]

**Pull request merged**